src/mpbuild/build_api.py

```diff
@@ -0,0 +1,288 @@
+                f"Entry port='{self.board.port.name}' missing in 'FIRMWARE_FILENAMES'!"
+            )
+
+        _filename = self.build_folder / filename
+        if not _filename.is_file():
+            raise MpbuildException(
+                f"The firmware for {self.board.name}-{self.variant} was not found: {_filename}"
+            )
+
+        return _filename
+
+    @property
+    def micro_version_text(self) -> str:
+        """
+        Example: '3.4.0; MicroPython v1.24.0 on 2024-10-25'
+        """
+        filename_mpversion_h = self.build_folder / "genhdr" / "mpversion.h"
+
+        repo_folder = self.build_folder.parents[2]
+        filename_modsys_c = repo_folder / "py" / "modsys.c"
+
+        def get_regex(filename: Path, pattern: re.Pattern) -> str:
+            try:
+                text = filename.read_text()
+            except FileNotFoundError as e:
+                raise MpbuildException(
+                    f"Firmware for {self.board.name}-{self.variant}: Could not read: {filename}"
+                ) from e
+
+            match = pattern.search(text)
+            if match is None:
+                raise MpbuildException(
+                    f"Firmware for {self.board.name}-{self.variant}: Could not find '{pattern.pattern}' in: {filename}"
+                )
+            return match.group(1)
+
+        git_tag = get_regex(filename_mpversion_h, self._REGEX_MICROPY_GIT_TAG)
+        build_date = get_regex(filename_mpversion_h, self._REGEX_MICROPY_BUILD_DATE)
+        modsys = get_regex(filename_modsys_c, self._REGEX_MICROPY_SYS_VERSION)
+        return f"{modsys}; MicroPython {git_tag} on {build_date}"
+
+
+def build(
+    logfile: Path,
+    board: Board,
+    variant: Optional[str] = None,
+    do_clean: bool = False,
+) -> Firmware:
+    """ """
+
+    build_cmd = docker_build_cmd(
+        board=board,
+        variant=variant,
+        do_clean=do_clean,
+        extra_args=[],
+        docker_interactive=False,
+    )
+
+    with logfile.open("w") as f:
+        begin_s = time.monotonic()
+        f.write(f"{build_cmd}\n\n\n")
+        f.flush()
+        proc = subprocess.run(
+            build_cmd,
+            shell=True,
+            check=False,
+            text=True,
+            stdout=f,
+            stderr=subprocess.STDOUT,
+        )
+        f.write(f"\n\nreturncode={proc.returncode}\n")
+        f.write(f"duration={time.monotonic()-begin_s:0.3f}s\n")
+
+    if proc.returncode != 0:
+        raise MpbuildDockerException(board=board, variant=variant, proc=proc)
+
+    build_folder = BuildFolder(board=board, variant=variant)
+
+    return Firmware(
+        filename=build_folder.firmware_filename,
+        board=board,
+        variant=variant,
+        micropython_version_text=build_folder.micro_version_text,
+    )
+
+
+def build_by_variant_normalized(
+    logfile: Path, db: Database, variant_normalized: str, do_clean: bool
+) -> Firmware:
+    """
+    This is the main entry point into mpbuild.
```

Comment:
This is not what I consider the main entry point to mpbuild and I don't think most users would see it that way either.

Reply:
Perhaps the `build()` function is the main entry point, and should be called "main entry point for the mpbuild package".

Reply:
Yes, that's what I'd expect.

Reply:
I removed the file build_api.py